internal/fields/validate.go

```diff
@@ -695,15 +695,17 @@ func createDocExpandingObjects(doc common.MapStr) (common.MapStr, error) {
 func isNumericKeyword(definition FieldDefinition, val interface{}) bool {
 	var isNumber bool
 	switch val := val.(type) {
-	case float64, []float64:
+	case bool, []bool, float64, []float64:
 		isNumber = true
 	case []interface{}:
 		isNumber = true
+	loop:
 		for _, v := range val {
-			_, ok := v.(float64)
-			if !ok {
+			switch v.(type) {
+			case bool, float64:
+			default:
 				isNumber = false
-				break
+				break loop
 			}
 		}
```

Comment:
Nit. I wonder if we can avoid the label.
```
		for _, v := range val {
			switch v.(type) {
			case bool, float64:
			default:
				isNumber = false
			}
			if !isNumber {
				break
			}
		}
```

Reply:
The generated code is worse for arguably less clear source. https://godbolt.org/z/fqrvEfrcW

Reply:
In general I consider the use of labels as less clear if there are alternatives. Here itself I had to think twice if this `loop:` was a special keyword for the `switch` that I didn't know about. But I guess this is a matter of opinions, so as you prefer :slightly_smiling_face: